### PR TITLE
Adds standalone-node simulation test

### PR DESCRIPTION
**Pull request by joeldudley**
The standalone nodes communicate via sockets and have standalone enclave servers that are connected to via RPC.

I know we're not worried about code duplication in the tests, but the linter was complaining so I created some shared logic for testing the simulations.

I also:

* Made the efficiency thresholds a parameter to the blockchain validation. This allows us to have different efficiencies for different test types (e.g. standalone vs in-memory)
* Renamed the `ObscuroInMemNodes` field on `Simulation` to `ObscuroNodes`, since they aren't necessarily in-memory

### CODE REVIEW

File: integration/simulation/simulation_tester.go [hunk 1]
```diff
@@ -6,23 +6,20 @@ import (
...
-// The unit of time is the "avgBlockDurationUSecs" - which is the average time between L1 blocks, which are the carriers of rollups.
-// Everything else is reported to this value. This number has to be adjusted in conjunction with the number of nodes. If it's too low,
-// the CPU usage will be very high during the simulation which might result in inconclusive results.
-func TestInMemoryMonteCarloSimulation(t *testing.T) {
+type createNetworkFunc = func(nrNodes int, avgGossipPeriod uint64, avgBlockDurationUSecs uint64, avgLatency uint64, stats *Stats) ([]*ethereummock.Node, []*host.Node)
+
+// testSimulation encapsulates the shared logic for simulating and testing various types of nodes.
+func testSimulation(t *testing.T, avgBlockDurationUSecs uint64, createNetwork createNetworkFunc, efficiencies EfficiencyThresholds) {
 	// define core test parameters
 	numberOfNodes := 10
```
Comment (tudor-malene) [inline integration/simulation/simulation_tester.go:18]: I'd say to move all these core parameters out of there as well. Maybe create a structure for them

Comment (tudor-malene) [reply integration/simulation/simulation_tester.go:18]: which would make the "createNetworkFunc" cleaner as well:

```type createNetworkFunc = func(testParams *TestParams, stats *Stats) ...```

File: integration/simulation/utils.go [hunk 2]
```diff
@@ -53,6 +61,28 @@ func createInMemObscuroNode(id int64, genesis bool, avgGossipPeriod uint64, avgB
 	return &node
 }
 
+func createStandaloneObscuroNode(id int64, genesis bool, avgGossipPeriod uint64, stats *Stats, peerAddrs []string) *host.Node {
```
Comment (tudor-malene) [inline integration/simulation/utils.go:64]: these are not really standalone. (started in a separate process)
It's more like the communication is more realistic

**Review by tudor-malene — CHANGES_REQUESTED** [review]
Thanks.
It's ok in principle. A few minor things

File: integration/simulation/utils.go (hunk 2 above)
Comment (joeldudley) [reply integration/simulation/utils.go:64]: Will fix, thanks

**Pull request merged**